packages/@aws-cdk/aws-location-alpha/lib/geofence-collection.ts

```diff
@@ -0,0 +1,168 @@
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as kms from 'aws-cdk-lib/aws-kms';
+import { ArnFormat, IResource, Lazy, Resource, Stack, Token } from 'aws-cdk-lib/core';
+import { Construct } from 'constructs';
+import { CfnGeofenceCollection } from 'aws-cdk-lib/aws-location';
+import { generateUniqueId } from './util';
+
+/**
+ * A Geofence Collection
+ */
+export interface IGeofenceCollection extends IResource {
+  /**
+   * The name of the geofence collection
+   *
+   * @attribute
+   */
+  readonly geofenceCollectionName: string;
+
+  /**
+   * The Amazon Resource Name (ARN) of the geofence collection resource
+   *
+   * @attribute Arn, CollectionArn
+   */
+  readonly geofenceCollectionArn: string;
+}
+
+/**
+ * Properties for a geofence collection
+ */
+export interface GeofenceCollectionProps {
+  /**
+   * A name for the geofence collection
+   *
+   * Must be between 1 and 100 characters and contain only alphanumeric characters,
+   * hyphens, periods and underscores.
+   *
+   * @default - A name is automatically generated
+   */
+  readonly geofenceCollectionName?: string;
+
+  /**
+   * A description for the geofence collection
+   *
+   * @default - no description
+   */
+  readonly description?: string;
+
+  /**
+   * The customer managed to encrypt your data.
+   *
+   * @default - Use an AWS managed key
+   * @see https://docs.aws.amazon.com/location/latest/developerguide/encryption-at-rest.html
+   */
+  readonly kmsKey?: kms.IKey;
+}
+
+abstract class GeofenceCollectionBase extends Resource implements IGeofenceCollection {
```

Comment:
What's the reasoning behind creating a base class when it's only inherited by one class?

Reply:
This is because the already implemented PlaceIndex class has a base class. I don't know if more classes will be added in the future, but I'm doing this to maintain consistency within the package. 
I'd appreciate your opinion on this if you have any.

https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-location-alpha/lib/place-index.ts#L95

Reply:
I personally don't see the purpose of having a separate base class. I would recommend to just use one class instead. Otherwise the PR LGTM.

Reply:
@GavinZZ 
Thanks. I've removed the based class.
Would you please review again?